hub/core/dataset.py

```diff
@@ -241,14 +240,37 @@ def _create_hidden_tensor(
         hash_samples: Optional[bool] = False,
         **kwargs,
     ):
+        """Creates a new hidden tensor in the dataset. # Hidden tensors are used to separate tensors created by the user from 
```

Comment:
Can you mention that these don't show up in dataset.tensors but can be accessed by getitem/getattr?